src/components/Video/Video.tsx

```diff
@@ -0,0 +1,133 @@
+  useState,
+  useEffect,
+  useRef
+} from 'react';
+import styles from './Video.module.scss';
+import { VideoData } from '../../types';
+import LazyLoad from '../LazyLoad/LazyLoad';
+
+import videojs from 'video.js';
+import 'video.js/dist/video-js.css';
+
+export type VideoProps = {
+  hasControls: boolean,
+  isAutoplaying: boolean
+  isLooping: boolean,
+  isMuted: boolean,
+  isPlaying?: boolean,
+  video: VideoData,
+};
+
+const Video = ({ isAutoplaying = true, hasControls = true, isPlaying = true, isLooping = true, isMuted = true, video }: VideoProps) => {
+  const videoRef = useRef(null);
+  const playerRef = useRef(null);
+  const isFirstLoad = useRef(true);
+
+  const [hasLoaded, setHasLoaded] = useState(false);
+  const [isMounted, setIsMounted] = useState(false);
+
+
+  if (!video) {
+    return <div className={`${styles.frame} ${styles.brokenVideo}`} />
+  }
+
+
+  const onPlayerReady = (player) => {
+
+    // TODO: Add custom buttons on the player (download)
+
+    // let downloadButtonElement = document.createElement('button')
+    // downloadButtonElement.innerHTML = '⤓';
+
+    // console.log(player.controlBar);
+
+    // player.controlBar.el().insertBefore(downloadButtonElement, player.controlBar.getChild('FullscreenToggle').el());
+
+    // player.addClass('vjs-vjsdownload');
+  };
+
+  useEffect(() => {
+
+    if (isFirstLoad.current && videoRef.current && video) {
+      isFirstLoad.current = false;
+      const videoElement = document.createElement("video-js");
+
+      videoElement.classList.add('vjs-big-play-centered');
+      videoRef.current.appendChild(videoElement);
+
+      const player = playerRef.current = videojs(videoElement, {
+        sources: [{
+          src: video.hls,
+          type: "application/x-mpegURL"
+        }],
+        autoplay: isAutoplaying,
+        muted: isMuted,
+        controls: hasControls,
+        fluid: true,
+        controlBar: {
+          pictureInPictureToggle: false, //firefox
+          subsCapsButton: false //safari
+        },
+        loop: isLooping,
+      }, () => onPlayerReady(player));
+
+      // Only consider the video loaded after it starts playing
+      player.on('progress', () => {
+        if (!hasLoaded) {
+          setHasLoaded(true)
+        }
+      })
+
+      // TODO: Connect to shared muted state
+      player.on('volumechange', () => {
+        console.log(player.muted())
+      })
+    }
+  }, [isMounted]);
+
+  useEffect(() => {
+    if (playerRef.current) {
+      playerRef.current.muted(isMuted);
+    }
+  }, [isMuted])
+
+  useEffect(() => {
+    const player = playerRef.current;
+
+    return () => {
+      if (player && !player.isDisposed()) {
+        player.dispose();
+        playerRef.current = null;
+      }
+    };
+  }, [playerRef]);
+
+  const hasLoadedClassName = hasLoaded ? styles.isLoaded : '';
+
+  const aspectRatioStyle = { aspectRatio: video.width / video.height }
+
+  return (
+    <LazyLoad callback={() => setIsMounted(true)}>
```

Comment:
Ah, I see now why we added this callback.

From what I can see, we only actually use `isMounted` to toggle one of the elements within LazyLoad when we've entered the viewport. This is what the `hasIntersected` prop exposed by the component does, we can use that directly here.